libeos-parental-controls/app-filter.c

```diff
@@ -452,3 +483,510 @@ epc_get_app_filter_finish (GAsyncResult  *result,
+  GAsyncResult *set_oars_filter_result;  /* (nullable) (owned) */
+} SetAppFilterData;
+
+static void
+set_app_filter_data_free (SetAppFilterData *data)
+{
+  epc_app_filter_unref (data->app_filter);
+  g_clear_object (&data->set_app_filter_result);
+  g_clear_object (&data->set_oars_filter_result);
+  g_free (data);
+}
+
+G_DEFINE_AUTOPTR_CLEANUP_FUNC (SetAppFilterData, set_app_filter_data_free)
+
+/**
+ * epc_set_app_filter_async:
+ * @connection: (nullable): a #GDBusConnection to the system bus, or %NULL to
+ *    use the default
+ * @user_id: ID of the user to set the filter for, typically coming from getuid()
+ * @app_filter: (transfer none): the app filter to set for the user
+ * @allow_interactive_authorization: %TRUE to allow interactive polkit
+ *    authorization dialogues to be displayed during the call; %FALSE otherwise
+ * @callback: a #GAsyncReadyCallback
+ * @cancellable: (nullable): a #GCancellable, or %NULL
+ * @user_data: user data to pass to @callback
+ *
+ * Asynchronously set the app filter settings for the given @user_id to the
+ * given @app_filter instance. This will set all fields of the app filter.
+ *
+ * @connection should be a connection to the system bus, where accounts-service
+ * runs. It’s provided mostly for testing purposes, or to allow an existing
+ * connection to be re-used. Pass %NULL to use the default connection.
+ *
+ * On failure, an #EpcAppFilterError, a #GDBusError or a #GIOError will be
+ * returned. The user’s app filter settings will be left in an undefined state.
+ *
+ * Since: 0.1.0
+ */
+void
+epc_set_app_filter_async (GDBusConnection     *connection,
+                          uid_t                user_id,
+                          EpcAppFilter        *app_filter,
+                          gboolean             allow_interactive_authorization,
+                          GCancellable        *cancellable,
+                          GAsyncReadyCallback  callback,
+                          gpointer             user_data)
+{
+  g_autoptr(GDBusConnection) connection_owned = NULL;
+  g_autoptr(GTask) task = NULL;
+  g_autoptr(SetAppFilterData) data = NULL;
+
+  g_return_if_fail (connection == NULL || G_IS_DBUS_CONNECTION (connection));
+  g_return_if_fail (app_filter != NULL);
+  g_return_if_fail (app_filter->ref_count >= 1);
+  g_return_if_fail (cancellable == NULL || G_IS_CANCELLABLE (cancellable));
+
+  task = g_task_new (NULL, cancellable, callback, user_data);
+  g_task_set_source_tag (task, epc_set_app_filter_async);
+
+  data = g_new0 (SetAppFilterData, 1);
+  data->user_id = user_id;
+  data->app_filter = epc_app_filter_ref (app_filter);
+  data->allow_interactive_authorization = allow_interactive_authorization;
+  g_task_set_task_data (task, g_steal_pointer (&data),
+                        (GDestroyNotify) set_app_filter_data_free);
+
+  if (connection == NULL)
+    g_bus_get (G_BUS_TYPE_SYSTEM, cancellable,
+               set_bus_cb, g_steal_pointer (&task));
```

Comment:
`get_bus_cb`

Reply:
Nope, this is meant to be `set_bus_cb()`, as it has the `*_set_*` continuations in it. I acknowledge the function naming is perverse and could be changed.

Reply:
Changed to `set_app_filter_get_bus_cb()`.